x-pack/plugins/security_solution/public/management/pages/policy/view/protection_updates/hooks/use_post_protection_updates_note.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { PROTECTION_UPDATES_NOTE_QUERY_KEY } from './use_get_protection_updates_note';
+import { useKibana } from '../../../../../../common/lib/kibana';
+import { resolvePathVariables } from '../../../../../../common/utils/resolve_path_variables';
+import { PROTECTION_UPDATES_NOTE_ROUTE } from '../../../../../../../common/endpoint/constants';
+
+interface UseProtectionUpdatesNote {
+  policyId: string;
+}
+
+interface NoteResponse {
+  note: string;
+}
+
+export const useCreateProtectionUpdatesNote = ({ policyId }: UseProtectionUpdatesNote) => {
+  const { http } = useKibana().services;
+  const queryClient = useQueryClient();
+
+  return useMutation<
+    { data: NoteResponse },
+    { body: { error: string; message: string } },
+    NoteResponse
+  >(
+    (payload) =>
+      http.post(resolvePathVariables(PROTECTION_UPDATES_NOTE_ROUTE, { policy_id: policyId }), {
+        version: '2023-10-31',
+        body: JSON.stringify(payload),
+      }),
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries([PROTECTION_UPDATES_NOTE_QUERY_KEY]);
```

Comment:
I believe this is how react-query work, `post` and `get` get the same query key so that once `post` is succesfful what is being consumed by `get` useHook in the component gets invalidated and another get is being sent. At least that is how I understand it :) and that's the reason they both need to share the same key. Updated key to be more verbose.